drivers/i2c/i2c_ite_enhance.c

```diff
@@ -85,6 +86,31 @@ enum i2c_ch_status {
 	I2C_CH_WAIT_NEXT_XFER,
 };
 
+#ifdef CONFIG_I2C_IT8XXX2_CQ_MODE
+struct i2c_host_cq_buffer {
+	/* Command queue tx payload. */
+	uint8_t i2c_cq_mode_tx_dlm[CONFIG_I2C_CQ_MODE_MAX_PAYLOAD_SIZE] __aligned(4);
+	/* Command queue rx payload. */
+	uint8_t i2c_cq_mode_rx_dlm[CONFIG_I2C_CQ_MODE_MAX_PAYLOAD_SIZE] __aligned(4);
+};
+#endif
+#ifdef CONFIG_I2C_TARGET
+/*
+ * When accessing data exceeds the maximum buffer, the actual reload address
+ * is one byte more than the maximum buffer size. Therefore, it is necessary to
+ * have a buffer in place to prevent overwriting other memory.
+ */
+#define PROTECT_MEM_BUF 4
+struct i2c_target_dma_buffer {
+	/* Target mode DMA output buffer. */
+	uint8_t __aligned(4)
+		out_buffer[CONFIG_I2C_TARGET_MAX_BUFFER_SIZE + PROTECT_MEM_BUF];
```

Comment:
This Kconfig being used only in the ite driver makes me believe it should be part of the driver configuration rather than i2c configuration

Reply:
Done